backend/inspirehep/records/api/literature.py

```diff
@@ -191,18 +192,24 @@ def update_refs_to_conferences(data):
                 continue
             conference["conference_record"] = get_ref_from_pid("con", pid)
 
+    def _deduplicate_references(self, data):
+        data["references"] = dedupe_list(data["references"])
+
     def update(
         self,
         data,
         disable_external_push=False,
         disable_relations_update=False,
+        deduplicate_references=True,
         *args,
         **kwargs,
     ):
         with db.session.begin_nested():
             LiteratureRecord.update_authors_signature_blocks_and_uuids(data)
             LiteratureRecord.update_refs_to_conferences(data)
             data = self.add_files(data)
+            if deduplicate_references and "references" in data:
+                self._deduplicate_references(data)
```

Comment:
isn't there a better place to put this (closer to reference extraction), instead of adding it to the general update API, where it's not needed in most cases?

Reply:
you mean in record editor? we decided that we don't want to do it there